jobs.groovy

```diff
@@ -544,93 +550,29 @@ def build_tabs(dashBranchName) {
 def define_job(dashProject, dashBranchName, branchName, job_type, job_name,
                full_name, job_values, module, isReleaseBuild) {
 
-    if (job_type in ['run_trial', 'run_trial_for_storage_driver', 'run_acceptance']) {
-        assert module != null
-        job(full_name) {
-            parameters {
-                // we pass the 'MODULE' parameter as the flocker module to test with trial
+    job(full_name) {
```

Comment:
The only remaining difference between the job types was modules_to_delete, this reads
that from build.yaml, so we now just loop over all the jobs and use the values, largely
ignoring the type.

The type is still used for cron jobs, but that could likely be unified as well. Keeping the
job type in the build.yaml may be useful for human understanding though.